lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2063,6 +2067,22 @@ void TypeChecker::checkOwnershipAttr(VarDecl *var, OwnershipAttr *attr) {
 
     diagnose(var->getStartLoc(), D, (unsigned) ownershipKind, underlyingType);
     attr->setInvalid();
+    return;
```

Comment:
This `return` doesn't really make sense when we're about to explicitly check for invalid and return below. Either all of the conditions in this function should be changed (in which case you don't need the "else" in `else if`), or you shouldn't worry about it. I'd lean towards the latter since it's less of a change from what's there.